apm-agent-common/src/test/java/co/elastic/apm/agent/util/VersionTest.java

```diff
@@ -19,31 +19,49 @@
 package co.elastic.apm.agent.util;
 
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
 class VersionTest {
 
-    @Test
-    void testVersion() {
-        assertThat(Version.of("1.2.3")).isEqualByComparingTo(Version.of("1.2.3"));
-        assertThat(Version.of("1.2.3-SNAPSHOT")).isEqualByComparingTo(Version.of("1.2.3"));
-        assertThat(Version.of("4.5.13")).isEqualByComparingTo(Version.of("4.5.13"));
-        assertThat(Version.of("4.5.13.redhat-00001")).isEqualByComparingTo(Version.of("4.5.13"));
-        assertThat(Version.of("httpclient-4.5.13")).isEqualByComparingTo(Version.of("4.5.13"));
-        assertThat(Version.of("httpclient-4.5.13.redhat-00001")).isEqualByComparingTo(Version.of("4.5.13"));
-        assertThat(Version.of("httpclient.4.5.13.redhat-00001")).isEqualByComparingTo(Version.of("4.5.13"));
-        assertThat(Version.of("httpclient.4.5.13.redhat")).isEqualByComparingTo(Version.of("4.5.13"));
-        assertThat(Version.of("httpclient.4.5.13-redhat")).isEqualByComparingTo(Version.of("4.5.13"));
+    @ParameterizedTest
+    @CsvSource(value = {
+        "1.0.0,                          <, 1.0.1",
+        "1.2.3,                          =, 1.2.3",
+        "1.2.3-SNAPSHOT,                 <, 1.2.3",
+        "4.5.13,                         =, 4.5.13",
+        "4.5.13.redhat-00001,            <, 4.5.13",
+        "httpclient-4.5.13,              =, 4.5.13",
+        "httpclient-4.5.13.redhat-00001, <, 4.5.13",
+        "httpclient.4.5.13.redhat-00001, =, 4.5.13.redhat-00001",
+        "httpclient.4.5.13.redhat,       <, 4.5.13",
```

Comment:
Why? I am not sure about the rationale of this

Reply:
Any suffix is considered lower than no suffix, as it usually means a pre-release. But yeah, in this case, we probably want `httpclient.4.5.13.redhat = 4.5.13`, right? Because that's important for determining whether or not to instrument a library.

Reply:
> Any suffix is considered lower than no suffix, as it usually means a pre-release

`Any` doesn't fit well with `usually`, because of the exceptions 🙂 .
The thing is that suffix is not necessarily `RC1` or `SNAPSHOT`, in which case ordering will be arbitrary. Moreover, as you say - for the main purpose we use it for (determine behavior based on version), `1.2.3.RC1 == 1.2.3`.

Reply:
Our Version class supports both semver and non-semver-compliant versions now. Semver specifies the [ordering of versions](https://semver.org/#spec-item-11), which we comply with.

If a version is known to potentially not be semver compliant, such as 4.5.13.redhat, we can just call `Version::withoutSuffix` and rely on only comparing the version numbers.